packages/@aws-cdk/aws-ec2/lib/machine-image.ts

```diff
@@ -311,3 +312,90 @@ export enum OperatingSystemType {
   LINUX,
   WINDOWS,
 }
+
+/**
+ * A machine image whose AMI ID will be searched using DescribeImages.
+ *
+ * The most recent, available, launchable image matching the given filter
+ * criteria will be used. Looking up AMIs may take a long time; specify
+ * as many filter criteria as possible to narrow down the search.
+ *
+ * The AMI selected will be cached in `cdk.context.json` and the same value
+ * will be used on future runs. To refresh the AMI lookup, you will have to
+ * evict the value from the cache using the `cdk context` command. See
+ * https://docs.aws.amazon.com/cdk/latest/guide/context.html for more information.
+ */
+export class LookupMachineImage implements IMachineImage {
```

Comment:
I believe we need to get this documented under the 'Context Methods' section in [here](https://docs.aws.amazon.com/cdk/latest/guide/context.html)

Reply:
Good catch.